packages/yew/tests/hydration.rs

```diff
@@ -540,3 +540,299 @@ async fn hydration_nested_suspense_works() {
+        html! {
+            <Suspense>
+                if number % 3 == 0 {
+                    <SuspendedNumber {number}/>
+                } else {
+                    <Number {number}/>
+                }
+            </Suspense>
+        }
+    }
+
+    #[hook]
+    pub fn use_suspend() -> SuspensionResult<()> {
+        let s = use_state(|| Suspension::from_future(async {}));
+
+        if s.resumed() {
+            Ok(())
+        } else {
+            Err((*s).clone())
+        }
+    }
+
+    let s = ServerRenderer::<App>::new().render().await;
+
+    gloo::utils::document()
+        .query_selector("#output")
+        .unwrap()
+        .unwrap()
+        .set_inner_html(&s);
+
+    sleep(Duration::ZERO).await;
+
+    Renderer::<App>::with_root(gloo_utils::document().get_element_by_id("output").unwrap())
+        .hydrate();
+
+    // Wait until all suspended components becomes revealed.
+    sleep(Duration::ZERO).await;
+    sleep(Duration::ZERO).await;
+    sleep(Duration::ZERO).await;
+    sleep(Duration::ZERO).await;
+
+    let result = obtain_result_by_id("output");
+    assert_eq!(
+        result.as_str(),
+        // Until all components become revealed, there will be component markers.
+        // As long as there's no component markers all components have become unsuspended.
+        r#"<div>0</div><div>1</div><div>2</div><div>3</div><div>4</div><div>5</div><div>6</div><div>7</div><div>8</div><div>9</div>"#
+    );
+}
+
+#[wasm_bindgen_test]
+async fn hydration_suspense_no_flickering() {
+    #[function_component(App)]
+    pub fn app() -> Html {
+        let fallback = html! { <h1>{"Loading..."}</h1> };
+        html! {
+            <Suspense {fallback}>
+                <Suspended/>
+            </Suspense>
+        }
+    }
+
+    #[derive(Properties, PartialEq, Clone)]
+    struct NumberProps {
+        number: u32,
+    }
+
+    #[function_component(SuspendedNumber)]
+    fn suspended_number(props: &NumberProps) -> HtmlResult {
+        use_suspend()?;
+
+        Ok(html! {
+            <Number ..{props.clone()}/>
+        })
+    }
+    #[function_component(Number)]
+    fn number(props: &NumberProps) -> Html {
+        html! {
+            <div>
+                {props.number.to_string()}
+            </div>
+        }
+    }
+
+    #[function_component(Suspended)]
+    fn suspended() -> HtmlResult {
+        use_suspend()?;
+
+        Ok(html! {
+            { for (0..10).map(|number|
+                html! {
+                    <SuspendedNumber {number}/>
+                }
+            )}
+        })
+    }
+
+    #[hook]
+    pub fn use_suspend() -> SuspensionResult<()> {
+        let s = use_state(|| {
+            Suspension::from_future(async {
+                gloo::timers::future::sleep(std::time::Duration::from_millis(50)).await;
+            })
+        });
+
+        if s.resumed() {
+            Ok(())
+        } else {
+            Err((*s).clone())
+        }
+    }
```

Comment:
Nit, (feel free to ignore): this seems like a good candidate to be replaced with `use_future`